test/workflow-orchestration.ts

```diff
@@ -660,4 +658,103 @@ describe('When a request spans multiple CMR pages', function () {
       });
     });
   });
+
+  describe('and the number of granules returned by the CMR changes while paging', function () {
+    const aggregateService = 'bar';
+    const nonAggregateService = 'foo';
+    const initialNumInputGranules = 5000;
+    const finalNumInputGranules = 10000;
+    const initialQueryCmrWorkItemCount = Math.ceil(initialNumInputGranules / env.cmrMaxPageSize);
+    const finalQueryCmrWorkItemCount = Math.ceil(finalNumInputGranules / env.cmrMaxPageSize);
+
+    hookServersStartStop();
+
+    before(async function () {
+      await truncateAll();
+      const job = buildJob({ numInputGranules: initialNumInputGranules });
+      await job.save(db);
+      this.jobID = job.jobID;
+
+      await buildWorkflowStep({
+        jobID: job.jobID,
+        serviceID: 'harmonyservices/query-cmr:latest',
+        stepIndex: 1,
+        workItemCount: initialQueryCmrWorkItemCount,
+      }).save(db);
+
+      await buildWorkflowStep({
+        jobID: job.jobID,
+        serviceID: nonAggregateService,
+        stepIndex: 2,
+        workItemCount: initialNumInputGranules,
+        hasAggregatedOutput: false,
+      }).save(db);
+
+      await buildWorkflowStep({
+        jobID: job.jobID,
+        serviceID: aggregateService,
+        stepIndex: 3,
+        workItemCount: 1,
+        hasAggregatedOutput: true,
+      }).save(db);
+
+      await buildWorkItem({
+        jobID: job.jobID,
+        serviceID: 'harmonyservices/query-cmr:latest',
+        workflowStepIndex: 1,
+        scrollID: '123abc',
+      }).save(db);
+    });
+    after(async function () {
+      await truncateAll();
+    });
+
+    describe('while retrieving granules from the CMR', function () {
+      it('sets the initial numInputGranules on the job', async function () {
+        const job = await Job.byJobID(db, this.jobID);
+        expect(job.numInputGranules).equals(initialNumInputGranules);
+      });
+      it('sets the initial number of work items for each step', async function () {
+        const workflowSteps = await getWorkflowStepsByJobId(db, this.jobID);
+        expect(workflowSteps[0].workItemCount).equals(initialQueryCmrWorkItemCount);
+        expect(workflowSteps[1].workItemCount).equals(initialNumInputGranules);
+        expect(workflowSteps[2].workItemCount).equals(1);
+      });
+
+      describe('when the CMR hits changes', async function () {
+
+        before(async function () {
+          const res = await getWorkForService(this.backend, 'harmonyservices/query-cmr:latest');
+          const { workItem } = JSON.parse(res.text);
+          workItem.status = WorkItemStatus.SUCCESSFUL;
+          workItem.hits = finalNumInputGranules;
+          workItem.results = [
```

Comment:
These files don't exist anymore - there's a new way to pass back the catalogs - take a look at the updates to the other tests in this file.

Reply:
Fixed